### ISSUE TITLE
Mpv renders video at 2x container's frame rate

### ISSUE THREAD

**Issue by anonmonogatari** [mpv-player/mpv#4476]
I don't know if  "render" is the correct term for this problem but mpv produces 50FPS while watching 24FPS video also produces 60FPS while watching 30FPS video. It also uses graphic processor way more than quicktime. I also added iStat Menus's frame per second info screen at the Sample files section.

### mpv version and platform
mpv 0.25.0 (C) 2000-2017 mpv/MPlayer/mplayer2 projects
OS X El Capitan 10.11.6 (15G1510)

### Reproduction steps

Open any video with mpv.

### Expected behavior

Mpv renders at video container's fps.

### Actual behavior

Mpv renders video at ~50FPS. (Nearly 2x. I say 2x because it also renders 30FPS video at 60FPS.)

### Log file

http://sprunge.us/jJWd

### Sample files
On the right quick time plays same video at real fps and on the left mpv plays video at 50FPS.
![frame](https://cloud.githubusercontent.com/assets/29133736/26710210/effa9e5e-4760-11e7-8447-f49e92e26ecb.png)

**Comment by Cpuroast** [comment]
Do you happen to have "interpolation=yes" in mpv.conf?

**Comment by anonmonogatari** [comment]
##Config
http://sprunge.us/fcGF

**Comment by Cpuroast** [comment]
Actually just tested on the same HD4000 video card using 10.12.5 and the same mpv 0.25 version with zero user-config and yes iStat Menus reports double the frame rate.

I think it's completely intended behaviour, either it's mpv or macOS' window manager is doing it.

I'm not registering as much GPU load, but that's because I only have the integrated 1680x1050 screen where as you have at least have an external 2560x1440 screen and the internal screen which is obviously more work for the older HD4000 GPU.

**Comment by Cpuroast** [comment]
Just so you know, you can drop these 2 lines:

vo=opengl -> No need to specify it, already the default on macOS
framedrop=vo -> No need to specify it, already the default

**Comment by Cpuroast** [comment]
Also tried with a GT650M, same double framerate result with no user config.

**Comment by anonmonogatari** [comment]
Well I don't think it's the intended behaviour. Older mpv version does not behave same.

https://laboratory.stolendata.net/~djinn/mpv_osx/

I downloaded version 0.21 from here and it produces 25FPS. Also, other players that uses libmpv such as [IINA](https://github.com/lhc70000/iina/releases/tag/v0.0.9) (btw this build based on mpv 0.25) doesn't have this problem.

Thx for the config advice btw.

**Comment by Akemi** [comment]
> I downloaded version 0.21 from here and it produces 25FPS.

there is quite a big gap between 0.21 and 0.25. it would be more helpful if you could tell us with what release it started.

**Comment by wm4** [comment]
@anonmonogatari does the GPU use go down with `--video-sync=display-resample`, stay the same, or does it get higher?

**Comment by Akemi** [comment]
i have a feeling the culprit is the early flush. does `--opengl-early-flush=no` fix it?

**Comment by anonmonogatari** [comment]
@wm4 `--video-sync=display-resample` makes mpv produce display fps which is 60 in my case (containers fps does not matter). 

@Akemi `--opengl-early-flush=no` fixes the problem.

**Comment by Cpuroast** [comment]
Personally, I'd use:

video-sync=display-resample
interpolation=yes

This would turn on interpolation and would actually make your "tscale=catmull_rom" directive do something as tscale only matters when using interpolation.
https://mpv.io/manual/stable/#options-tscale

**Comment by anonmonogatari** [comment]
@Cpuroast 
`video-sync=display-resample
interpolation=yes`

Is not always a viable option especially if your on battery and have a hidpi screen.

**Comment by Akemi** [comment]
k, then it's loosely related to #3739. don't expect a fix for this any time soon, though i think the workaround should be sufficient enough for now.

**Comment by pcordes** [comment]
@Akemi: I started seeing this on Arch Linux when going from mpv `1:0.26.0-3` to `1:0.27.0-5`.  vo=opengl, hwdec=vaapi (Intel Skylake GT2 integrated graphics, i7-6700k).

This bug introduces frame dropping when upscaling from 1280x720 to 1920x1080 with `scale=ewa_lanczossharp` / `cscale=ewa_lanczossharp` / `profile=opengl-hq`, for source FPS of 29.97.  OSD shows FPS as `59.94` when I hit `n`, with this keybind in `input.txt` (i.e. for `${estimated-vf-fps:}`):

```
     n show_text "${width}x${height}${?video-frame-info/interlaced==yes:i} ${estimated-vf-fps:} V ${video-bitrate:na} A ${audio-bitrate:na} ${video-frame-info/picture-type} (${=time-pos})"
```

It also introduces frame dropping with 1920x1080 video when I zoom (so the luma samples need scaling, not just chroma).  And it sucks for playing 50fps or 60fps video.

My OpenGL setup is the usual X11 Xorg server with Mesa 17.3, running KDE with compositing disabled for fullscreen windows.  (I forget, I may have disabled compositing all the time).

**Comment by Akemi** [comment]
fixed via c5e4538bc4b63f02bf89f62aa25a37b9eb0c0316.

**Comment by mr-berndt** [comment]
Since this ist three years old, I would assume this fix should be included in arch linux version of mpv.
Still I do have the error, playing back a 50 fps video runs at twice the speed, so is obviously being treated as a 25 fps file.

mpv correctly reports 50 fps when started from commandline but dose play at the wrong speed.

It is kind of hard to imagine I am the only one seeing this.
